Shared.Tests/IntegrationTestBase.cs

```diff
@@ -168,16 +226,47 @@ protected IEnumerable<Resource> GetAllResults(Func<String, ListResourcesResult>
             return resources;
         }
 
+        protected string GetUniquePublicId()
+        {
+            return GetUniquePublicId(ResourceParameterType.upload);
+        }
+
+        protected virtual string GetUniquePublicId(ResourceParameterType actionType, string suffix="")
+        {
+            var publicId = $"{m_apiTest}_{m_publicIdsToClear[actionType].Count + 1}{suffix}";
+
+            m_publicIdsToClear[actionType].Add(publicId);
+
+            return publicId;
+        }
+
+        protected void AddCreatedPublicId(ResourceParameterType actionType, string publicId)
+        {
+            if (!string.IsNullOrEmpty(publicId))
+                m_publicIdsToClear[actionType].Add(publicId);
+        }
+
         [OneTimeTearDown]
-        public void Cleanup()
+        public virtual void Cleanup()
         {
-            string publicId = string.Format("TestForTagSearch_{0}", m_suffix);
-            DelResResult delResult = m_cloudinary.DeleteResources(new string[] { publicId });
-            publicId = string.Concat(m_suffix, "_TestForTagSearch");
-            delResult = m_cloudinary.DeleteResources(new string[] { publicId });
-            publicId = string.Concat(m_suffix, "_TestForSearch");
-            delResult = m_cloudinary.DeleteResources(new string[] { publicId });
-            m_cloudinary.DeleteResourcesByTag(m_test_tag);
+            m_cloudinary.DeleteResources(new DelResParams() { Tag = m_apiTag, ResourceType = ResourceType.Image });
+            m_cloudinary.DeleteResources(new DelResParams() { Tag = m_apiTag, ResourceType = ResourceType.Raw });
+            m_cloudinary.DeleteResources(new DelResParams() { Tag = m_apiTag, ResourceType = ResourceType.Video });
+            m_cloudinary.DeleteResources(
+                new DelResParams() { Tag = m_apiTag, ResourceType = ResourceType.Raw, Type = "private" });
+            m_cloudinary.DeleteResourcesByPrefix(m_folderPrefix);
+            m_cloudinary.DeleteResourcesByPrefix(m_apiTest);
+            m_cloudinary.DeleteResourcesByPrefix(m_apiTag);
+
+            foreach (var item in m_publicIdsToClear)
+            {
+                m_cloudinary.DeleteResources(new DelResParams()
+                {
+                    Type = item.Key.ToString(),
```

Comment:
"Type" here is the "storage type" see:
https://cloudinary.com/documentation/admin_api#browse_resources
some of the values from "ResourceParameterType" like "text", "sprite", "multi" don't make any sense here.
Please clarify definition and the main purpose of ResourceParameterType enum.

Reply:
Looks like the documentation page is outdated. Here’s a small test
Let's create a sprite in account [screenshot](https://www.dropbox.com/s/pycxxc4pfr9pn9f/Screenshot%202018-04-25%2001.19.28.png?dl=0)
Then try to delete it with default “upload” storage type
```bash
curl -g -X "DELETE" "https://xxx:xxx@api.cloudinary.com/v1_1/rtlstudio/resources/image/upload?public_ids[]=sample" 
```
It wouldn’t be deleted. But if we use “sprite” storage type 
```bash
curl -g -X "DELETE" "https://xxx:xxx@api.cloudinary.com/v1_1/rtlstudio/resources/image/sprite?public_ids[]=sample"
```
It would delete the sprite. Moreover, if we open an account media library - we have a dropdown with types:  [screenshot](https://www.dropbox.com/s/60r9wuykf8h3ka6/Screenshot%202018-04-25%2001.30.01.png?dl=0)
As you can see, we have way more options there than specified in docs.

The main purpose of `ResourceParameterType` (now renamed to `StorageType` to match docs) is to be able to delete resources from account by generated public ids. Without explicit storage type it would use a default value and don't delete non "upload" resources.